File: LinkedList/237-delete-node-in-a-linked-list/delete-node-in-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public void deleteNode(ListNode node) {
        ListNode temp = null;
        while(node.next!=null){
            int x = node.val;
            node.val = node.next.val;
            node.next.val = x;
            temp = node;
            node = node.next;
        }
        temp.next = null;
    }
}

File: Amazon/237-delete-node-in-a-linked-list/delete-node-in-a-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def deleteNode(self, node):
        """
        :type node: ListNode
        :rtype: void Do not return anything, modify node in-place instead.
        """
        temp = None

        while(node.next is not None):
            node.val = node.next.val
            temp = node
            node = node.next

        temp.next = None